mu/logic.py

```diff
@@ -234,6 +234,56 @@ def check_pycodestyle(code):
     return style_feedback
 
 
+def get_runtime_hex_path():
+    """
+    Returns the path to the hex runtime file - if this has been
+    specified under element 'microbit_runtime_hex' in settings.json.
+    This can be a fully-qualified file path, or just a file name
+    in which case the file should be located in the workspace directory.
+    """
+    runtime_hex_path = ""
+    settings_path = get_settings_path()
+    settings = {}
+    try:
+        with open(settings_path) as f:
+            settings = json.load(f)
+    except FileNotFoundError:
+        logger.error('Settings file {} does not exist.'.format(settings_path))
+    except ValueError:
+        logger.error('Settings file {} could not be parsed.'.format(
+                     settings_path))
+    else:
+        if 'microbit_runtime_hex' in settings and \
+                settings['microbit_runtime_hex'] != "":
+            runtime_hex_path = os.path.join(
```

Comment:
Agree with None instead of "" (I'm still an old C guy...)

Regarding checking whether the path exists: the reason I didn't do that here is because of this flow:

1. User puts a path in the settings.json but hasn't actually put the hex file at that location yet
2. They run Mu and flash
3. This function would then return None as the file specified in settings.json doesn't exist
4. They exit Mu and the None gets persisted to settings.json
5. They get round to putting the hex file in the right place and run Mu again, only to find the path magically disappeared from settings.json

Which I think might be annoying?

Maybe there is a way around that where we just don't touch whatever path they put in the settings.json even if it is garbage - but then I think we'd need some other way of reading it so it can be echoed back when the settings.json is recreated?